src/Database/Validator/Index.php

```diff
@@ -25,39 +34,128 @@ public function getDescription(): string
         return $this->message;
     }
 
+    /**
+     * @param Document $collection
+     * @return bool
+     */
+    public function checkEmptyIndexAttributes(Document $collection): bool
+    {
+        foreach ($collection->getAttribute('indexes', []) as $index) {
+            if (empty($index->getAttribute('attributes', []))) {
+                $this->message = 'No attributes provided for index';
+                return false;
+            }
+        }
+
+        return true;
+    }
+
+    /**
+     * @param Document $collection
+     * @return bool
+     * @throws DatabaseException
+     */
+    public function checkFulltextIndexNonString(Document $collection): bool
+    {
+        foreach ($collection->getAttribute('indexes', []) as $index) {
+            if ($index->getAttribute('type') === Database::INDEX_FULLTEXT) {
+                foreach ($index->getAttribute('attributes', []) as $attributeName) {
+                    $attribute = $this->attributes[$attributeName] ?? new Document([]);
+                    if ($attribute->getAttribute('type', '') !== Database::VAR_STRING) {
+                        $this->message = 'Attribute "'.$attribute->getAttribute('key', $attribute->getAttribute('$id')).'" cannot be part of a FULLTEXT index';
```

Comment:
Let's add some text here explaining that it must be a string attribute:

```suggestion
                        $this->message = 'Attribute "'.$attribute->getAttribute('key', $attribute->getAttribute('$id')).'" cannot be part of a FULLTEXT index, must be of type string';
```